internal/kafka/internal/services/util.go

```diff
@@ -70,7 +70,7 @@ func BuildKeycloakClientNameIdentifier(kafkaRequestID string) string {
 	return fmt.Sprintf("%s-%s", "kafka", strings.ToLower(kafkaRequestID))
 }
 
-func BuildCustomClaimCheck(kafkaRequest *dbapi.KafkaRequest, ssoconfigProvider keycloak.SSOProvider) string {
+func BuildCustomClaimCheck(kafkaRequest *dbapi.KafkaRequest, ssoconfigProvider string) string {
 	if ssoconfigProvider == keycloak.REDHAT_SSO {
 		return fmt.Sprintf("@.rh-org-id == '%s'|| @.org_id == '%s' || @.clientId == '%s'", kafkaRequest.OrganisationId, kafkaRequest.OrganisationId, kafkaRequest.CanaryServiceAccountClientID)
 	} else {
```

Comment:
If we keep `ssoconfigProvider` as object, this function could become a method of `SSOProvider`.

This way, code that needs the customClaim will be able to do something like this:
```go
customClaim := ssoconfigProvider.BuildCustomClaimCheck(kafkaRequest)
```

I'd prefer that approach, because that way we avoid the spreading of `if provider == xxx`

Reply:
I think this also applies to my comment above. If I am wrong could you clarify please? https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1007#discussion_r873779876